test/concrete/flowErc1155/FlowTest.sol

```diff
@@ -298,4 +300,98 @@ contract Erc1155FlowTest is FlowERC1155Test {
+            vm.mockCall(address(iTokenB), abi.encodeWithSelector(IERC20.transfer.selector), abi.encode(true));
+
+            vm.expectCall(
+                address(iTokenA),
+                abi.encodeWithSelector(IERC20.transferFrom.selector, alice, flowErc1155, erc20InAmmount),
+                2
+            );
+            vm.expectCall(address(iTokenB), abi.encodeWithSelector(IERC20.transfer.selector, alice, erc20OutAmmount), 2);
+        }
+
+        // Stack mint
+        {
+            ERC20Transfer[] memory erc20Transfers = new ERC20Transfer[](2);
+
+            erc20Transfers[0] =
+                ERC20Transfer({token: address(iTokenA), from: alice, to: address(flowErc1155), amount: erc20InAmmount});
+
+            erc20Transfers[1] =
+                ERC20Transfer({token: address(iTokenB), from: address(flowErc1155), to: alice, amount: erc20OutAmmount});
+
+            ERC1155SupplyChange[] memory mints = new ERC1155SupplyChange[](1);
+            mints[0] = ERC1155SupplyChange({account: alice, id: tokenId, amount: erc20InAmmount});
+
+            uint256[] memory stack = generateFlowStack(
+                FlowERC1155IOV1(
+                    mints,
+                    new ERC1155SupplyChange[](0),
+                    FlowTransferV1(erc20Transfers, new ERC721Transfer[](0), new ERC1155Transfer[](0))
+                )
+            );
+            interpreterEval2MockCall(stack, new uint256[](0));
+        }
+
+        {
+            vm.startPrank(alice);
+            flowErc1155.flow(evaluable, new uint256[](0), new SignedContextV1[](0));
+            vm.stopPrank();
+
+            assertEq(IERC1155(address(flowErc1155)).balanceOf(alice, tokenId), erc20InAmmount);
+        }
+
+        // Stack burn
+        {
+            ERC20Transfer[] memory erc20Transfers = new ERC20Transfer[](2);
+
+            erc20Transfers[0] =
+                ERC20Transfer({token: address(iTokenA), from: alice, to: address(flowErc1155), amount: erc20InAmmount});
+
+            erc20Transfers[1] =
+                ERC20Transfer({token: address(iTokenB), from: address(flowErc1155), to: alice, amount: erc20OutAmmount});
+
+            ERC1155SupplyChange[] memory burns = new ERC1155SupplyChange[](1);
+            burns[0] = ERC1155SupplyChange({account: alice, id: tokenId, amount: erc20OutAmmount});
+
+            uint256[] memory stack = generateFlowStack(
+                FlowERC1155IOV1(
+                    new ERC1155SupplyChange[](0),
+                    burns,
+                    FlowTransferV1(erc20Transfers, new ERC721Transfer[](0), new ERC1155Transfer[](0))
+                )
+            );
+
+            interpreterEval2MockCall(stack, new uint256[](0));
+        }
+
+        {
+            vm.startPrank(alice);
+            flowErc1155.flow(evaluable, new uint256[](0), new SignedContextV1[](0));
+            vm.stopPrank();
+
+            assertEq(IERC1155(address(flowErc1155)).balanceOf(alice, tokenId), erc20InAmmount - erc20OutAmmount);
```

Comment:
the comment for this test says "in exchange for" but i only see an assertion for one balance change